solver/src/solver/baseline_solver.rs

```diff
@@ -141,30 +146,29 @@ impl BaselineSolver {
         }
     }
 
-    fn solve(&self, liquidity: Vec<Liquidity>) -> Vec<Settlement> {
-        let (user_orders, amm_map) = liquidity.into_iter().fold(
-            (Vec::new(), HashMap::<_, Vec<_>>::new()),
-            |(mut user_orders, mut amm_map), liquidity| {
-                match liquidity {
-                    Liquidity::Limit(order) => user_orders.push(order),
-                    Liquidity::ConstantProduct(order) => {
-                        amm_map.entry(order.tokens).or_default().push(Amm {
-                            tokens: order.tokens,
-                            order: AmmOrder::ConstantProduct(order),
-                        });
-                    }
-                    Liquidity::WeightedProduct(order) => {
-                        for tokens in order.token_pairs() {
-                            amm_map.entry(tokens).or_default().push(Amm {
-                                tokens,
-                                order: AmmOrder::WeightedProduct(order.clone()),
+    fn solve(&self, user_orders: Vec<LimitOrder>, liquidity: Vec<Liquidity>) -> Vec<Settlement> {
+        let amm_map =
+            liquidity
+                .into_iter()
+                .fold(HashMap::<_, Vec<_>>::new(), |mut amm_map, liquidity| {
+                    match liquidity {
+                        Liquidity::ConstantProduct(order) => {
+                            amm_map.entry(order.tokens).or_default().push(Amm {
+                                tokens: order.tokens,
+                                order: AmmOrder::ConstantProduct(order),
                             });
                         }
+                        Liquidity::WeightedProduct(order) => {
+                            for tokens in order.token_pairs() {
+                                amm_map.entry(tokens).or_default().push(Amm {
+                                    tokens,
+                                    order: AmmOrder::WeightedProduct(order.clone()),
+                                });
```

Comment:
order is cloned in one place but not the other...

Reply:
This was already the case before (look at line 160 that got removed).

Additionally, this is needed here. The reason is you have 1 token pair per order for constant product orders, so no `clone` required, however you have n token pairs per order for weighted products, so you need to clone `n` times. Also worth noting, clippy should report on superfluous `clone` calls, so the linter should catch unnecessary clones.